plugins/deep-link/src/lib.rs

```diff
@@ -154,23 +172,87 @@ mod imp {
+                if config.desktop.contains_scheme(&url.scheme().to_string()) {
+                    return Some(url);
+                } else if cfg!(debug_assertions) {
+                    log::warn!("argument {url} does not match any configured deep link scheme; skipping it");
+                }
+            }
+        }
+
+        None
     }
 
     impl<R: Runtime> DeepLink<R> {
+        /// Checks if the provided list of arguments (which should match [`std::env::args`])
+        /// contains a deep link argument (for Linux and Windows).
+        ///
+        /// On Linux and Windows the deep links trigger a new app instance with the deep link URL as its only argument.
+        ///
+        /// This function does what it can to verify if the argument is actually a deep link, though it could also be a regular CLI argument.
+        /// To enhance its checks, we only match deep links against the schemes defined in the Tauri configuration
+        /// i.e. dynamic schemes WON'T be processed.
+        ///
+        /// This function updates the [`Self::get_current`] value and emits a `deep-link://new-url` event.
+        #[cfg(desktop)]
+        pub fn handle_cli_arguments<S: AsRef<str>, I: Iterator<Item = S>>(&self, mut args: I) {
+            use tauri::Emitter;
+
+            let Some(config) = &self.config else {
+                return;
+            };
+
+            if let Some(url) = deep_link_from_args(config, args) {
+                let mut current = self.current.lock().unwrap();
+                current.replace(vec![url.clone()]);
+                let _ = self.app.emit("deep-link://new-url", vec![url]);
+            }
+        }
+
         /// Get the current URLs that triggered the deep link. Use this on app load to check whether your app was started via a deep link.
         ///
         /// ## Platform-specific:
         ///
-        /// - **Windows / Linux**: Unsupported, will return [`Error::UnsupportedPlatform`](`crate::Error::UnsupportedPlatform`).
+        /// - **Windows / Linux**: This function reads the command line arguments and checks if there's only one value, which must be an URL with scheme matching one of the configured values.
+        ///     Note that you must manually check the arguments when registering deep link schemes dynamically with [`Self::register`].
+        ///     Additionally, the deep link might have been provided as a CLI argument so you should check if its format matches what you expect.
         pub fn get_current(&self) -> crate::Result<Option<Vec<url::Url>>> {
-            #[cfg(not(any(windows, target_os = "linux")))]
             return Ok(self.current.lock().unwrap().clone());
-            #[cfg(any(windows, target_os = "linux"))]
-            Err(crate::Error::UnsupportedPlatform)
+        }
+
+        /// Registers all schemes defined in the configuration file.
+        ///
+        /// This is useful to ensure the schemes are registered even if the user unregistered it
```

Comment:
This feels wrong tbh and we shouldn't recommend that at all. If a user unregistered a protocol they should be in control of that and we shouldn't keep annoying them.

Reply:
I know that app images and portable builds on Windows need to manually register that but we either warn against this or provide some checks that user chose to disable this protocol, so you shouldn't re-register them

Reply:
oops oops oops :joy: